brayns/io/loaders/volume/VolumeLoader.cpp

```diff
@@ -19,182 +19,222 @@
 
 #include "VolumeLoader.h"
 
-#include <brayns/engine/Model.h>
-#include <brayns/engine/Scene.h>
-#include <brayns/engine/Volume.h>
-#include <brayns/utils/StringUtils.h>
+#include <brayns/engine/components/Volumes.h>
+#include <brayns/engine/systems/GenericBoundsSystem.h>
+#include <brayns/engine/systems/VolumeCommitSystem.h>
+#include <brayns/engine/systems/VolumeInitSystem.h>
+#include <brayns/engine/volume/types/RegularVolume.h>
+
+#include <brayns/utils/FileReader.h>
+#include <brayns/utils/parsing/Parser.h>
+#include <brayns/utils/string/StringExtractor.h>
+#include <brayns/utils/string/StringSplitter.h>
+#include <brayns/utils/string/StringTrimmer.h>
 
 #include <filesystem>
-#include <fstream>
-#include <map>
-#include <sstream>
-#include <string>
+#include <unordered_map>
 
-namespace brayns
-{
 namespace
 {
-template<size_t M, typename T>
-std::string to_string(const glm::vec<M, T> &vec)
+class MhdParser
 {
-    std::stringstream stream;
-    stream << vec;
-    return stream.str();
-}
+public:
+    static std::unordered_map<std::string, std::string> parse(const std::string &filename)
+    {
+        auto content = brayns::FileReader::read(filename);
+        auto view = std::string_view(content);
+
+        auto result = std::unordered_map<std::string, std::string>();
+
+        auto ctr = 1ul;
+        while (!view.empty())
+        {
+            auto line = brayns::StringExtractor::extractLine(view);
+            brayns::StringExtractor::extract(view, 1);
+
+            auto keyAndValue = brayns::StringSplitter::split(line, "=");
+            if (keyAndValue.size() != 2)
+            {
+                throw std::runtime_error("Could not parse line " + std::to_string(ctr));
+            }
+
+            auto key = brayns::StringTrimmer::trim(keyAndValue[0]);
+            auto value = brayns::StringTrimmer::trim(keyAndValue[1]);
+            result.emplace(key, value);
+        }
 
-template<typename T>
-glm::vec<3, T> parseVec3(const std::string &str, std::function<T(std::string)> conv)
+        return result;
+    }
+};
+
+class MhdRequiredKeys
```

Comment:
Just personal preference but I would parse the header as a struct if all these keys are mandatory and raise the errors from here.

Reply:
I think its superflous on this scenario, since the result is used to fill the Blob to load the volume using the RawLoader.